vulcanai/datasets/tabulardataset.py

```diff
@@ -130,14 +133,17 @@ def __getitem__(self, idx):
         # the label column, iloc gets the row, then access values and convert
 
         if self.label_column:
-            xs = self.df.drop(self.label_column,
-                              axis=1).iloc[[2]].values.tolist()[0]
-            xs = torch.tensor(xs, dtype=torch.float)
-            y = self.df[[self.label_column]].iloc[[idx]].values.tolist()[0][0]
+            if self.all_xs is None:
+                self.all_xs = self.df.drop(self.label_column,
+                              axis=1)
+            if self.all_y is None:
+                self.all_y = self.df[[self.label_column]]
+            xs = torch.from_numpy(self.all_xs.iloc[[idx]].values[0]).float()
+            y = self.all_y.iloc[[idx]].values.tolist()[0][0]
             y = torch.tensor(y, dtype=torch.long)
```

Comment:
why not also do a `torch.from_numpy` here as well?

Reply:
because it was giving me all sorts of weird values due to it being a single value (shape was wrong). doesn't matter anyways though because we're going to scrap all of this once I  push the new tabulardataset.